cli/azd/pkg/azure/arm_template.go

```diff
@@ -104,6 +104,10 @@ type AzdMetadataType string
 
 const AzdMetadataTypeLocation AzdMetadataType = "location"
 const AzdMetadataTypeGenerate AzdMetadataType = "generate"
+const AzdMetadataTypePrincipalLogin AzdMetadataType = "principalLogin"
+const AzdMetadataTypePrincipalId AzdMetadataType = "principalId"
+const AzdMetadataTypePrincipalType AzdMetadataType = "principalType"
+const AzdMetadataTypeIpAddress AzdMetadataType = "ipAddress"
```

Comment:
I'm very nervous about this feature (and also the implementation) - Are we totally hosed if we don't have it?  I assume this is for us to be able to scope some firewall rules to "the current user's machine" but I am very nervous about corner cases in the implementation with respect to things like NAT.